src/main/java/graphql/schema/AbstractGraphQLType.java

```diff
@@ -0,0 +1,32 @@
+package graphql.schema;
+
+import static graphql.Assert.assertNotNull;
+
+public class AbstractGraphQLType implements GraphQLType {
```

Comment:
I understand, usually I would've turned `GraphQLType` into an abstract class to at least keep the name property and equals/hashcode in the same place, because this really is common behavior of all types, but I didn't want to make that incompatible change all over the code base either.

If you prefer I can easily push the property and method implementations down again.